### PR TITLE
update wifi module doc types to str

### PR DESCRIPTION
**Pull request by FoamyGuy**
If I understand correctly this is the resolution that was discussed here: https://github.com/adafruit/Adafruit_CircuitPython_Typing/pull/5

Changed all of the ReadableBuffer typed arguments in `wifi` to `str`

### CODE REVIEW

File: shared-bindings/wifi/Radio.c [hunk 1]
```diff
@@ -226,8 +226,8 @@ STATIC mp_obj_t wifi_radio_stop_station(mp_obj_t self) {
 MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_stop_station_obj, wifi_radio_stop_station);
 
 //|     def start_ap(self,
-//|                  ssid: ReadableBuffer,
-//|                  password: ReadableBuffer = b"",
+//|                  ssid: str,
+//|                  password: str = b"",
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.c:230]: The given default value is a `bytes`, not a `str`.
```suggestion
//|                  password: str = "",
```

File: shared-bindings/wifi/Radio.c [hunk 2]
```diff
@@ -304,11 +304,11 @@ STATIC mp_obj_t wifi_radio_stop_ap(mp_obj_t self) {
 MP_DEFINE_CONST_FUN_OBJ_1(wifi_radio_stop_ap_obj, wifi_radio_stop_ap);
 
 //|     def connect(self,
-//|                 ssid: ReadableBuffer,
-//|                 password: ReadableBuffer = b"",
+//|                 ssid: str,
+//|                 password: str = b"",
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.c:308]: The given default value is a `bytes`, not a `str`.
```suggestion
//|                 password: str = "",
```

File: shared-bindings/wifi/Radio.c [hunk 3]
```diff
@@ -304,11 +304,11 @@ STATIC mp_obj_t wifi_radio_stop_ap(mp_obj_t self) {
...
 //|                 *,
 //|                 channel: Optional[int] = 0,
-//|                 bssid: Optional[ReadableBuffer] = b"",
+//|                 bssid: Optional[str] = b"",
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.c:311]: The given default value is a `bytes`, not a `str`.
```suggestion
//|                 bssid: Optional[str] = "",
```

File: shared-bindings/wifi/Radio.c [hunk 4]
```diff
@@ -149,7 +149,7 @@ const mp_obj_property_t wifi_radio_mac_address_obj = {
                MP_ROM_NONE },
 };
 
-//|     mac_address_ap: ReadableBuffer
+//|     mac_address_ap: str
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.c:152]: `ReadableBuffer` is correct here. It's not a human-readable string, it's the bytes of the MAC address.
```suggestion
//|     mac_address_ap: ReadableBuffer
```

File: shared-bindings/wifi/Radio.c [hunk 5]
```diff
@@ -117,7 +117,7 @@ const mp_obj_property_t wifi_radio_hostname_obj = {
               MP_ROM_NONE},
 };
 
-//|     mac_address: ReadableBuffer
+//|     mac_address: str
```
Comment (dhalbert) [inline shared-bindings/wifi/Radio.c:120]: `ReadableBuffer` is correct here. It's not a human-readable string, it's the bytes of the MAC address.
```suggestion
//|     mac_address: ReadableBuffer
```

**Review by dhalbert — CHANGES_REQUESTED** [review]
I've looked at the low-level stuff in `ports/espressif/common-hal/wifi/Radio.c`, and it's not so easy. The ESP-IDF routines take `uint8_t` for the ssid and password, and take const char for the hostname. I don't know if they assume ASCII or not, or just "some bytes".

For the string-like values, we may want to convert them to `bytes()` before passing them down, and describe them as `Union[str | ReadableBuffer]`. The fact that you can pass either right now is an accident of MicroPython treating `str`s as obeying the buffer protocol.

File: shared-bindings/wifi/Radio.c (hunk 2 above)
Comment (anecdata) [reply shared-bindings/wifi/Radio.c:308]: SSID can be composed of any byte values, not only those that can be represented in a string.

**Comment by FoamyGuy** [comment]
Latest commit changes `mac_address` and `mac_address_ap` back to ReadableBuffer. Also change the `str` types to be `Union[str | ReadableBuffer]` and remove the bytes `b` from the default values that had it.

I think these are the changes that you suggested, but let me know if I've misinterpreted something.

I wasn't  sure what you meant by "convert them to bytes() before passing them down" Does it mean like on this line: https://github.com/FoamyGuy/circuitpython/blob/6a792ab373bff7e47adcaffaca02019506037f20/shared-bindings/wifi/Radio.c#L291

it would wrap that value in bytes() call like this?
```
common_hal_wifi_radio_start_ap(self, bytes(ssid.buf), ssid.len, bytes(password.buf), password.len, args[ARG_channel].u_int, authmode, args[ARG_max_connections].u_int);
```

**Comment by dhalbert** [comment]
> wasn't sure what you meant by "convert them to bytes() before passing them down" Does it mean like on this line: https://github.com/FoamyGuy/circuitpython/blob/6a792ab373bff7e47adcaffaca02019506037f20/shared-bindings/wifi/Radio.c#L291
> 
> it would wrap that value in bytes() call like this?
> 
> ```
> common_hal_wifi_radio_start_ap(self, bytes(ssid.buf), ssid.len, bytes(password.buf), password.len, args[ARG_channel].u_int, authmode, args[ARG_max_connections].u_int);
> ```

Ah, OK, it is already passing raw bytes down, so nothing needs to be done. We can't do `bytes(ssid.buf)`, etc. -- that is strictly a Python construct.

**Review by dhalbert — APPROVED** [review]
OK, I think this is good as it stands. Thanks!

**Pull request merged**